### PR TITLE
Fix the duplicate entries when connection, instance, and provider account deleted, added DBaaS QuickStart Version 

### PR DESCRIPTION
**Pull request by redhatHameed**
Fixed below issues 
https://issues.redhat.com/browse/DBAAS-904
https://issues.redhat.com/browse/DBAAS-902
https://issues.redhat.com/browse/DBAAS-905
https://issues.redhat.com/browse/DBAAS-907
and 
https://issues.redhat.com/browse/DBAAS-912

Signed-off-by: Abdul Hameed <ahameed@redhat.com>

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: controllers/metrics.go [hunk 1]
```diff
@@ -184,17 +184,13 @@ func SetInventoryMetrics(inventory dbaasv1alpha1.DBaaSInventory, execution Execu
 func setInventoryStatusMetrics(inventory dbaasv1alpha1.DBaaSInventory) {
 	for _, cond := range inventory.Status.Conditions {
 		if cond.Type == dbaasv1alpha1.DBaaSInventoryReadyType {
+			DBaaSInventoryStatusGauge.DeletePartialMatch(prometheus.Labels{metricLabelProvider: inventory.Spec.ProviderRef.Name, metricLabelName: inventory.Name, metricLabelNameSpace: inventory.Namespace})
```
Comment (jianrongzhang89) [inline controllers/metrics.go:187]: What about removing metricLabelProvider here so that it is consistent with other places?

Comment (redhatHameed) [reply controllers/metrics.go:187]: it works both ways, keeping here will not harm, in other cases user can delete the provider account before deleting the connection and instance and provider account passed empty to delete metrics and deletion not happened successful.

Comment (jianrongzhang89) [reply controllers/metrics.go:187]: OK

**Review by jianrongzhang89 — APPROVED** [review]
LGTM

**Review by jeremyary — APPROVED** [review]

**Pull request merged**